src/sonic_ax_impl/mibs/ietf/rfc2737.py

```diff
@@ -318,8 +318,34 @@ def reinit_data(self):
         self.physical_name_map[chassis_mgmt_sub_id] = name
         self.physical_fru_map[chassis_mgmt_sub_id] = self.NOT_REPLACEABLE
 
+        exceptions = []
+        has_runtime_err = False
+        # Catch exception in the iteration
+        # This makes sure if any exception is raised in the mid of loop
+        # every updater's reinit_data function will be always called
+        # So that the redis subscriptions always get chance to be cleaned,
+        # Otherwise if the exception never recover,
+        # the redis subscription keeps increasing but never got consumed,
+        # this causes redis memory leak.
         for updater in self.physical_entity_updaters:
-            updater.reinit_data()
+            try:
+                updater.reinit_data()
+            except BaseException as e:
+                mibs.logger.error(f"Jianquan {e}, type{type(e)}")
```

Comment:
type on error log

Reply:
Debug log for unit test, removed, thanks for pointing out!